src/main/java/hudson/plugins/jobConfigHistory/JobConfigHistory.java

```diff
@@ -118,28 +123,22 @@ public class JobConfigHistory extends Plugin {
      */
     private boolean showChangeReasonCommentWindow = true;
 
-    @Override
-    public void start() throws Exception {
+    @DataBoundConstructor
+    @Restricted(NoExternalUse.class)
+    public JobConfigHistory() {
         load();
-        loadRegexpPatterns();
     }
 
     @Override
-    public void configure(StaplerRequest req, JSONObject formData)
-            throws IOException, ServletException, FormException {
-
-        historyRootDir = formData.getString("historyRootDir").trim();
-        setMaxHistoryEntries(formData.getString("maxHistoryEntries").trim());
-        setMaxDaysToKeepEntries(formData.getString("maxDaysToKeepEntries").trim());
-        setMaxEntriesPerPage(formData.getString("maxEntriesPerPage").trim());
-        skipDuplicateHistory = formData.getBoolean("skipDuplicateHistory");
-        excludePattern = formData.getString("excludePattern");
-        saveModuleConfiguration = formData.getBoolean("saveModuleConfiguration");
-        showBuildBadges = formData.getString("showBuildBadges");
-        excludedUsers = formData.getString("excludedUsers");
-        showChangeReasonCommentWindow = formData.getBoolean("showChangeReasonCommentWindow");
+    public boolean configure(StaplerRequest req, JSONObject formData) {

```

Comment:
According to JCasC's [REQUIREMENTS.md](https://github.com/jenkinsci/configuration-as-code-plugin/blob/configuration-as-code-1.55.1/docs/REQUIREMENTS.md#rule-1-dont-write-code-for-data-binding), `BulkChange` can be used to avoid multiple `save()` calls.
But as the don't use `save()` in this plugin, this might not be required.

Reply:
Right that's my understanding too. That `save()` in setters can be use instead with `BulkChange` in `configure()`. I can do that as per the docs.

Reply:
Done.